### PR TITLE
changed await to asynchronous comm

### PR DESCRIPTION
**Pull request by ealight**

### CODE REVIEW

File: src/main/scala/study/softserve/scala/Application.scala [hunk 1]
```diff
@@ -1,32 +1,39 @@
...
 
     system.scheduler.schedule(0 millis, schedulerTimeout minute) {
       println("--------------------------------------------------")
-      val weatherContainer = for (city <- cities) yield {
-        val result = Await.result(creator ? city, timeout.duration)
-        result.asInstanceOf[CityWeather]
+      val futures = for (city <- cities) yield actors ? city
+
+      system.scheduler.scheduleOnce(apiResponseTimeout seconds) {
+        val cont = futures
```
Comment (ikachur) [inline src/main/scala/study/softserve/scala/Application.scala]: What is `cont` here? Better use the full name

File: src/main/scala/study/softserve/scala/Application.scala [hunk 2]
```diff
@@ -1,32 +1,39 @@
...
+          .flatMap(future =>
+            future
+              .mapTo[CityWeather]
+              .value
+              .map(x => x.get))
```
Comment (ikachur) [inline src/main/scala/study/softserve/scala/Application.scala:31]: Get is unsafe to use, consider `getOrElse`

File: src/main/scala/study/softserve/scala/Application.scala [hunk 3]
```diff
@@ -1,32 +1,39 @@
...
+
+        val sorted = cont.sortBy(_.main.temp)
```
Comment (ikachur) [inline src/main/scala/study/softserve/scala/Application.scala]: Please override `compare` method of the class. Then just use `cont.sorted`

File: src/main/scala/study/softserve/scala/RequestActor.scala [hunk 4]
```diff
@@ -28,14 +28,15 @@ class RequestActor extends Actor {
...
+
+      responseFuture
         .flatMap(_.entity.toStrict(2 seconds))
         .map(_.data.utf8String)
         .map(response => parse(response)
           .camelizeKeys
           .extract[CityWeather])
-
-      sender() ! Await.result(future, timeout.duration)
+        .onComplete(response => sen ! response.get)
```
Comment (ikachur) [inline src/main/scala/study/softserve/scala/RequestActor.scala]: Again, get is not safe
Another way (without callback)
`<Future>.pipeTo(sender())`

File: src/main/scala/study/softserve/scala/package.scala [hunk 5]
```diff
@@ -10,9 +10,10 @@ package object scala {
   implicit val formats: DefaultFormats.type = DefaultFormats
 
 
+
```
Comment (ikachur) [inline src/main/scala/study/softserve/scala/package.scala]: So many extra lines :) Please run `format` in the IDE.

File: src/main/scala/study/softserve/scala/package.scala [hunk 6]
```diff
@@ -10,9 +10,10 @@ package object scala {
...
   val openWeatherApiKey: String = "932990025e60573ec55fb150d6ae8368"
-  val cities = List("London", "Paris", "Kyiv", "Lviv", "Singapore", "Rome", "Beijing", "Seoul")
+  val cities = List("London", "Paris", "Kyiv", "Lviv", "Singapore", "Rome", "Beijing", "Seoul", "Milan")
 
-  val apiResponseTimeout = 15
+  val apiResponseTimeout = 3
```
Comment (ikachur) [inline src/main/scala/study/softserve/scala/package.scala:14]: Is this seconds?

Comment (ealight) [reply src/main/scala/study/softserve/scala/package.scala:14]: yes, i increased it to 15 to test problem with response

> Is this seconds?

**Pull request merged**